packages/typescript-fetch-runtime/src/request-bodies/url-search-param.spec.ts

```diff
@@ -0,0 +1,185 @@
+      // note: undefined by spec
+      object: "R=100&G=200&B=150",
+    },
+    {
+      explode: false,
+      style: "spaceDelimited",
+      // note: undefined by spec
+      string: "color=blue",
+      array: "color=blue+black+brown",
+      // todo: original example from oas differs, https://github.com/OAI/OpenAPI-Specification/issues/4813
+      // array: "color=blue%20black%20brown",
+      object: "color=R+100+G+200+B+150",
+      // todo: original example from oas differs, https://github.com/OAI/OpenAPI-Specification/issues/4813
+      // object: "color=R%20100%20G%20200%20B%20150",
+    },
+    {
+      explode: true,
+      style: "pipeDelimited",
+      // note: undefined by spec
+      string: "color=blue",
+      // note: undefined by spec
+      array: "color=blue&color=black&color=brown",
+      // note: undefined by spec
+      object: "R=100&G=200&B=150",
+    },
+    {
+      explode: false,
+      style: "pipeDelimited",
+      // note: undefined by spec
+      string: "color=blue",
+      array: "color=blue%7Cblack%7Cbrown",
+      object: "color=R%7C100%7CG%7C200%7CB%7C150",
+    },
+    {
+      explode: true,
+      style: "deepObject",
+      // note: undefined by spec
+      string: "color=blue",
+      // note: undefined by spec; using stripe expectation of `color[0]=blue&color[1]=black&color[2]=brown`
+      array: "color%5B0%5D=blue&color%5B1%5D=black&color%5B2%5D=brown",
+      object: "color%5BR%5D=100&color%5BG%5D=200&color%5BB%5D=150",
+    },
+    {
+      explode: false,
+      style: "deepObject",
+      // note: undefined by spec
+      string: "color=blue",
+      // note: undefined by spec
+      array: "color=blue%2Cblack%2Cbrown",
+      // note: undefined by spec, using JSON.stringify representation
+      object: "color=%7B%22R%22%3A100%2C%22G%22%3A200%2C%22B%22%3A150%7D",
+    },
+  ] as const)(
+    "oas 4.8.12.4 style examples - when explode: $explode and style: $style",
+    ({explode, style, string, array, object}) => {
+      it("serializes a string correctly", () => {
+        const actual = requestBodyToUrlSearchParams(
+          {color: "blue"},
+          {color: {explode, style}},
+        )
+        expect(actual.toString()).toStrictEqual(string)
+      })
+
+      it("serializes a array correctly", () => {
+        const actual = requestBodyToUrlSearchParams(
+          {color: ["blue", "black", "brown"]},
+          {color: {explode, style}},
+        )
+        expect(actual.toString()).toStrictEqual(array)
+      })
+
+      it("serializes a object correctly", () => {
+        const actual = requestBodyToUrlSearchParams(
+          {color: {R: 100, G: 200, B: 150}},
+          {
+            color: {
+              explode,
+              style,
+            },
+          },
+        )
+        expect(actual.toString()).toStrictEqual(object)
+      })
+    },
+  )
+
+  describe("stripe /v1 api conventions; style: 'deepObject' explode: true", () => {
```

Comment:
Where behavior is otherwise undefined in terms of OAI, I've aligned to what the stripe v1 API does/expects. It's a popular API, the only example of nested objects/arrays I have right now.